### PR TITLE
[Log] fix spdlog init race

### PR DESCRIPTION
**Pull request by ashione**
## Related issue number

There is init race when we create multiple default loggers in threads. To fix this, it's suggested to make default logger as a singleton instance.
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by zhongchun — APPROVED** [review]
LGTM.

**Review by jovany-wang — APPROVED** [review]

File: src/ray/util/logging.cc [hunk 1]
```diff
@@ -96,22 +96,19 @@ class SpdLogMessage final {
...
-    }
+  inline std::shared_ptr<spdlog::logger> GetDefaultLogger() {
     // We just emit all log informations to stderr when no default logger has been created
     // before starting ray log, which is for glog compatible.
-    if (!logger) {
-      logger = spdlog::stderr_color_mt("stderr");
-      logger->set_pattern(RayLog::GetLogFormatPattern());
-    }
+    static auto logger = spdlog::stderr_color_mt("stderr");
+    logger->set_pattern(RayLog::GetLogFormatPattern());
```
Comment (ashione) [inline src/ray/util/logging.cc:103]: Somebody may say `set_pattern` could not be thread-safe. To simplify to fix this issue, this pr will gurantee that get stderr logger is safe, but I found set_pattern is low cost and thread-safe made in spdlog source code 
```cpp
// Set global formatter. Each sink in each logger will get a clone of this object
SPDLOG_INLINE void registry::set_formatter(std::unique_ptr<formatter> formatter)
{
    std::lock_guard<std::mutex> lock(logger_map_mutex_);
    formatter_ = std::move(formatter);
    for (auto &l : loggers_)
    {
        l.second->set_formatter(formatter_->clone());
    }
}


```

**Comment by rkooo567** [comment]
@ashione This PR has tons of windows failures. Are they related?

**Comment by rkooo567** [comment]
Also please ping me when it is ready to go

**Comment by ashione** [comment]
> Also please ping me when it is ready to go

I‘ll refine this code and test internally and push new commits to this PR this week later.  cc @jovany-wang

**Comment by ashione** [comment]
> @ashione This PR has tons of windows failures. Are they related?

I'm not sure why so many windows failures, but seems unrelated to these changes.

**Comment by rkooo567** [comment]
Yeah it is disastrous in our master now.... I think they should be unrelated.

**Pull request merged**